plugins/src/main/java/com/google/fhir/gateway/plugin/SmartFhirPermissionChecker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.fhir.gateway.plugin;
+
+import com.google.fhir.gateway.plugin.SmartFhirScope.Permission;
+import com.google.fhir.gateway.plugin.SmartFhirScope.PrincipalContext;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SmartFhirPermissionChecker {
```

Comment:
nit: Thanks for adding this abstraction. To help with verbosity of Java, let's try to use shorter names where we can, e.g., in this case I think `SamrtPermissionChecker` or `SmartScopeChecker` might be good enough (also in the variable names of this type).